hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/StoreScanner.java

```diff
@@ -1202,5 +1322,24 @@ public void shipped() throws IOException {
       trySwitchToStreamRead();
     }
   }
+
+  final boolean checkIfScannersToBeReset() {
+    // to prevent volatile boolean scannersToBeReset lookup if
+    // isStoreFileForceArchivalEnabled is false already
+    if (!this.store.isStoreFileForceArchivalEnabled()) {
+      return false;
```

Comment:
Here, we are trying to prevent volatile boolean lookup if the config `hbase.hstore.file.force.archive` is by default turned off.

FYI @anoopsjohn @ramkrish86 @apurtell